packages/@react-spectrum/menu/src/MenuDialogTrigger.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, SlotProvider, useIsMobileDevice} from '@react-spectrum/utils';
+import helpStyles from '@adobe/spectrum-css-temp/components/contextualhelp/vars.css';
+import {ItemProps} from '@react-types/shared';
+import {MenuDialogContext, useMenuStateContext} from './context';
+import {Modal, Popover} from '@react-spectrum/overlays';
+import React, {Key, ReactElement, useRef} from 'react';
+import {useOverlayTriggerState} from '@react-stately/overlays';
+
+function MenuDialogTrigger<T>(props: ItemProps<T> & {isUnavailable?: boolean, targetKey: Key}): ReactElement {
+  let {isUnavailable} = props;
+
+  let {state: menuState} = useMenuStateContext();
+  let state = useOverlayTriggerState({isOpen: menuState.expandedKeys.has(props.targetKey), onOpenChange: (val) => {
+    if (!val) {
+      if (menuState.expandedKeys.has(props.targetKey)) {
+        menuState.toggleKey(props.targetKey);
+      }
+    }
+  }});
+  let slots = {};
+  if (isUnavailable) {
```

Comment:
What other use cases are there for MenuDialogTrigger other than the unavailable menu item? Wondering if we need the `isUnavailable` prop at all

Reply:
People might put anything in here, maybe a Picker, or a filtering list of options with a textfield. so unfortunately we do need to specify what kind it is

Reply:
Hrm, should those components be supported within the MenuDialogTrigger use case wise? Feels like we are giving the user a lot of freedom to add what ever they want to the sub dialog, but this was supposed to cover just a contextual help like experience. I guess explicit guidelines in the docs could help with that...

Will think further on the api, but this is ok for now

Reply:
yes, they could, and I don't know that we want to discourage it explicitly because this would be how to do it

but I think if we just document the unavailable one, most people won't stray from it